src/Network/Freddy.hs

```diff
@@ -0,0 +1,48 @@
+{-# OPTIONS -XOverloadedStrings #-}
+module Network.Freddy (connect, respondTo) where
+
+import Network.AMQP
+import Data.Text (Text, pack)
+import Data.ByteString.Lazy.Char8 (ByteString)
+import Network.Freddy.Result
+
+connect :: String -> Text -> Text -> Text -> IO Connection
+connect host vhost user password =
+  openConnection host vhost user password
+
+respondTo :: Connection -> String -> (ByteString -> Either ByteString ByteString) -> IO ConsumerTag
```

Comment:
Not sure what do you mean with your last paragraph. My idea was that client will provide functions to handle DB, etc, and injects them where IO is possible (e.g https://github.com/indrekj/hubstation/blob/master/start.hs ). Then respondTo itself doesn't need to care about it. (I haven't tried integrating with db so I have no idea how it will actually look like).